packages/shared/scope-hoisting/src/link.js

```diff
@@ -595,14 +595,11 @@ export function link({
       importedFiles.set(filePath, imported);
     }
 
-    // If not unused, add the asset to the list of specifiers to import.
-    if (!isUnusedValue(ancestors) && mod.meta.exportsIdentifier) {
-      invariant(imported.assets != null);
-      imported.assets.add(mod);
+    invariant(imported.assets != null);
+    imported.assets.add(mod);
 
-      let initIdentifier = getIdentifier(mod, 'init');
-      return t.callExpression(initIdentifier, []);
-    }
+    let initIdentifier = getIdentifier(mod, 'init');
+    return t.callExpression(initIdentifier, []);
```

Comment:
This is the only change, the rest is removing unused function parameters.

What was the reason for the `mod.meta.exportsIdentifier` condition?

Reply:
Probably old...